GCE acceptance test fails with panic: interface conversion

I think this is a result of the change from a list of maps to a map in #1517.

/cc @phinze 
```
google_compute_instance_template.foobar
2015/04/24 11:17:59 [DEBUG] vertex root.google_compute_instance_template.foobar: walking
2015/04/24 11:17:59 [DEBUG] vertex root.google_compute_instance_template.foobar: evaluating
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalSequence
2015/04/24 11:17:59 [DEBUG] root: eval: terraform.EvalNoop
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalInstanceInfo
2015/04/24 11:17:59 [DEBUG] root: eval: terraform.EvalNoop
2015/04/24 11:17:59 [DEBUG] root: eval: terraform.EvalNoop
2015/04/24 11:17:59 [DEBUG] root: eval: terraform.EvalNoop
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalOpFilter
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalSequence
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalReadDiff
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalIf
2015/04/24 11:17:59 [DEBUG] root: eval: terraform.EvalNoop
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalIf
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalInterpolate
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalGetProvider
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalReadState
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalDiff
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalReadDiff
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalCompareDiff
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalGetProvider
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalReadState
2015/04/24 11:17:59 [DEBUG] root: eval: *terraform.EvalApply
2015/04/24 11:17:59 [DEBUG] apply: google_compute_instance_template.foobar: executing Apply
panic: interface conversion: interface is []interface {}, not map[string]interface {}

goroutine 1096 [running]:
github.com/hashicorp/terraform/builtin/providers/google.resourceInstanceMetadata(0xc2082f20c0, 0xc208378408)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/builtin/providers/google/resource_compute_instance.go:767 +0xa8
github.com/hashicorp/terraform/builtin/providers/google.resourceComputeInstanceTemplateCreate(0xc2082f20c0, 0x53f4c0, 0xc20826ffc0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/builtin/providers/google/resource_compute_instance_template.go:326 +0x2b6
github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc20802da00, 0xc2083a1260, 0xc20837d520, 0x53f4c0, 0xc20826ffc0, 0x13f401, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/helper/schema/resource.go:145 +0x2c1
github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc20800b7a0, 0xc2082269c0, 0xc2083a1260, 0xc20837d520, 0x1, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/helper/schema/provider.go:154 +0x1da
github.com/hashicorp/terraform/terraform.(*EvalApply).Eval(0xc208226b40, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval_apply.go:67 +0x62a
github.com/hashicorp/terraform/terraform.EvalRaw(0xb13c98, 0xc208226b40, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x254
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc208090ec0, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval_sequence.go:10 +0xf9
github.com/hashicorp/terraform/terraform.EvalRaw(0xb13440, 0xc208090ec0, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x254
github.com/hashicorp/terraform/terraform.(*EvalOpFilter).Eval(0xc2082f0f30, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval_filter_operation.go:37 +0x71
github.com/hashicorp/terraform/terraform.EvalRaw(0xb13468, 0xc2082f0f30, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x254
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc2080d2e40, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval_sequence.go:10 +0xf9
github.com/hashicorp/terraform/terraform.EvalRaw(0xb13440, 0xc2080d2e40, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x254
github.com/hashicorp/terraform/terraform.Eval(0xb13440, 0xc2080d2e40, 0xb132b0, 0xc208274cc0, 0x0, 0x0, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/eval.go:34 +0x79
github.com/hashicorp/terraform/terraform.func·027(0x5f2bc0, 0xc208092350, 0x0, 0x0)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/terraform/graph.go:158 +0x754
github.com/hashicorp/terraform/dag.func·006(0x5f2bc0, 0xc208092350, 0xc2083dcd20, 0xc2083dcd80)
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/dag/dag.go:215 +0x106
created by github.com/hashicorp/terraform/dag.(*AcyclicGraph).Walk
        /Users/dcarley/projects/gocode/src/github.com/hashicorp/terraform/dag/dag.go:224 +0x845
```

--- a/builtin/providers/google/resource_compute_instance_template.go
+++ b/builtin/providers/google/resource_compute_instance_template.go
@@ -126,12 +126,9 @@ func resourceComputeInstanceTemplate() *schema.Resource {
 			},
 
 			"metadata": &schema.Schema{
-				Type:     schema.TypeList,
+				Type:     schema.TypeMap,
 				Optional: true,
 				ForceNew: true,
-				Elem: &schema.Schema{
-					Type: schema.TypeMap,
-				},
 			},
 
 			"network": &schema.Schema{

